File: solutions/FilipTheFrog/solve/go/FilipTheFrogSolver.go
package main

import (
    "./FilipTheFrog"
    "fmt"
    "os"
    "topcoder"
)

func main() {
    if len(os.Args) < 3 {
        fmt.Fprintf(os.Stderr, "Usage: %s <input_file> <output_file>", os.Args[0])
        os.Exit(-1)
    }

    var (
        fileHandle *os.File
        errIO error
        positions []int
        L int
    )

    if fileHandle, errIO = os.Open(os.Args[1]); errIO != nil {
        panic(errIO)
    }
    topcoder.Read(fileHandle, &positions, &L)
    fileHandle.Close()

    if fileHandle, errIO = os.Create(os.Args[2]); errIO != nil {
        panic(errIO)
    }
    topcoder.Write(fileHandle, FilipTheFrog.CountReachableIslands(positions, L))
}

File: solutions/FilipTheFrog/solve/ruby/FilipTheFrogSolver.rb
#! /usr/bin/env ruby

require_relative "FilipTheFrog"

require "gettc/types"
include Gettc

def init()
    gettc_home = File.join(ENV["HOME"], ".gettc")
    if ENV.has_key?("GETTC_HOME")
        gettc_home = ENV["GETTC_HOME"]
    end 
    gettc_home = File.expand_path(gettc_home)
    $LOAD_PATH << File.join(gettc_home, "include", "ruby")
    require "topcoder"
    include TopCoder
end

def main()
    reader = Reader.new(IO.read(ARGV[0]))
    positions = reader.next(TArray.new(TInt))
    reader.next()
    L = reader.next(TInt)

    result = FilipTheFrog.new().countReachableIslands(positions, L)
    IO.write(ARGV[1], Writer.new().next(result, TInt).to_s())
end
init()
main()
